compiler-plugin/src/main/java/io/ballerina/stdlib/xmldata/compiler/XmldataRecordFieldValidator.java

```diff
@@ -0,0 +1,151 @@
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.stdlib.xmldata.compiler;
+
+import io.ballerina.compiler.syntax.tree.ExpressionNode;
+import io.ballerina.compiler.syntax.tree.ModuleVariableDeclarationNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.RecordFieldNode;
+import io.ballerina.compiler.syntax.tree.TypedBindingPatternNode;
+import io.ballerina.compiler.syntax.tree.VariableDeclarationNode;
+import io.ballerina.projects.plugins.AnalysisTask;
+import io.ballerina.projects.plugins.SyntaxNodeAnalysisContext;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticFactory;
+import io.ballerina.tools.diagnostics.DiagnosticInfo;
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Xmldata record field analyzer.
+ */
+public class XmldataRecordFieldValidator implements AnalysisTask<SyntaxNodeAnalysisContext> {
+
+    private final List<RecordFieldNode> recordNodes = new ArrayList<>();
+    private final List<RecordFieldNode> validatedRecordNodes = new ArrayList<>();
+    private static final String STRING = "string";
+    private static final String DECIMAL = "decimal";
+    private static final String FLOAT = "float";
+    private static final String BOOLEAN = "boolean";
+    private static final String INT = "int";
+    private static final String QUESTION_MARK = "?";
+    private static final String TO_RECORD = "xmldata:toRecord";
+    private static final String VERTICAL_BAR = "|";
+    private static final String SQUARE_BRACKET = "[]";
+    private static final String BRACKET = "[";
+
+    @Override
+    public void perform(SyntaxNodeAnalysisContext ctx) {
+        List<Diagnostic> diagnostics = ctx.semanticModel().diagnostics();
+        for (Diagnostic diagnostic : diagnostics) {
+            if (diagnostic.diagnosticInfo().severity() == DiagnosticSeverity.ERROR) {
+                return;
+            }
+        }
+        Node node = ctx.node();
+        if (node instanceof RecordFieldNode) {
+            this.recordNodes.add((RecordFieldNode) node);
+        }
+        if (node instanceof VariableDeclarationNode) {
+            VariableDeclarationNode variableDeclarationNode = (VariableDeclarationNode) node;
```

Comment:
Why are we casting `node` to `VariableDeclarationNode` if it is already an instance of `VariableDeclarationNode` type?